### PR TITLE
implements multi-client login management

### PR DESCRIPTION
**Pull request by wilybrace**
DBaaS relies on the use of concurrent user access to the Crunchy Bridge
API meaning that a single shared session was not sufficient for correct
functionality. This change preserves the local management of short-live
authentication tokens while allowing multiple clients to have active
"sessions" with the API.

Additional changes in this commit:
 - changes PersonalTeamID to DefaultTeamID, more generic name and more
   inline with future implementation detail changes
 - removes redundant log info in reconcile context logger - key/value
   information added to the logger was already present in the context
 - removes package level logger setup as part of the multi-user cleanup

### CODE REVIEW

**Comment by wilybrace** [comment]
Opened as draft to allow testing with this commit, new approach requires additional modifications to operator deployment yaml to include environment variables for creation controller.

**Comment by redhatHameed** [comment]
was trying to test but operator get terminated with below 
```
021-09-17T14:33:44.430-0400    ERROR   setup   error setting up Crunchy Bridge API client      {"error": "secrets \"crunchybridge_api_key\" not found"}
github.com/go-logr/zapr.(*zapLogger).Error
        /Users/ahameed/go/src/github.com/CrunchyData/crunchy-bridge-operator/vendor/github.com/go-logr/zapr/zapr.go:132
sigs.k8s.io/controller-runtime/pkg/log.(*DelegatingLogger).Error
        /Users/ahameed/go/src/github.com/CrunchyData/crunchy-bridge-operator/vendor/sigs.k8s.io/controller-runtime/pkg/log/deleg.go:144
main.main
        /Users/ahameed/go/src/github.com/CrunchyData/crunchy-bridge-operator/main.go:143
runtime.main
        /usr/local/go/src/runtime/proc.go:225
exit status 1
```

 the error is fine but it should continue the reconciler

**Comment by wilybrace** [comment]
I think this is tied to the need to specify the setup for the creation controller, I'll push a patch which disables that one if the environment variables aren't configured.

**Comment by wilybrace** [comment]
@redhatHameed - see if that works for you

**Comment by redhatHameed** [comment]
> @redhatHameed - see if that works for you

Thanks, it works now.

File: internal/bridgeapi/client.go [hunk 1]
```diff
@@ -32,51 +32,83 @@ var (
...
 	}
-	return client
+
+	if sess, err := sessionCache.GetSession(c.authTarget, cp, c.log); err != nil {
+		return nil, err
+	} else {
+		c.session = sess
+	}
+
+	fmt.Printf("Session state: %#v\n", c.session)
```
Comment (redhatHameed) [inline internal/bridgeapi/client.go:68]: replace with logger?

File: main.go [hunk 2]
```diff
@@ -116,65 +124,63 @@ func main() {
...
+		SecretField: keySecret,
+	}
+
+	runBridgeControllers := true
+
+	bridgeClient, err := bridgeapi.NewClient(apiURL, ksp, bridgeapi.SetLogger(setupLog))
+	if err != nil {
+		setupLog.Info("unable to configure Crunchy Bridge API client for crunchybridge controllers, disabling")
+		runBridgeControllers = false
+		// setupLog.Error(err, "error setting up Crunchy Bridge API client for crunchybridge controllers")
```
Comment (redhatHameed) [inline main.go:155]: remove these lines if not required?

**Review by redhatHameed — COMMENTED** [review]
@wilybrace Thanks only a few minor comments  else lgtm

**Pull request merged**